### PR TITLE
RUN-3142 Remove inter app messages from the logs

### PR DESCRIPTION
**Pull request by weiteho**
inter app bus message payload might contain sensitive data, removed it from log.

### CODE REVIEW

File: src/browser/api_protocol/transport_strategy/api_transport_base.ts [hunk 1]
```diff
@@ -64,4 +64,14 @@ export abstract class ApiTransportBase<T> {
             ackFunction(new NackPayload(err));
         };
     }
+
+    protected payloadReplacer(key: string, value: any): any {
+        if (key && key === 'payload') {
```
Comment (whyn07m3) [inline src/browser/api_protocol/transport_strategy/api_transport_base.ts]: can it just be `if (key === 'payload') {` ?

File: src/browser/api_protocol/transport_strategy/elipc_strategy.ts [hunk 2]
```diff
@@ -94,9 +94,15 @@ export class ElipcStrategy extends ApiTransportBase<MessagePackage> {
             };
 
             /* tslint:disable: max-line-length */
-            system.debugLog(1, `received in-runtime${data.isSync ? '-sync ' : ''}: ${e.frameRoutingId} [${identity.uuid}]-[${identity.name}] ${JSON.stringify(data)}`);
+            if (data.action && data.action === 'publish-message') {
```
Comment (whyn07m3) [inline src/browser/api_protocol/transport_strategy/elipc_strategy.ts]: can it just be `if (data.action === 'publish-message') {` ?

**Comment by weiteho** [comment]
yes, removed unnecessary check.

File: src/browser/api_protocol/transport_strategy/elipc_strategy.ts [hunk 3]
```diff
@@ -94,9 +94,15 @@ export class ElipcStrategy extends ApiTransportBase<MessagePackage> {
             };
 
             /* tslint:disable: max-line-length */
-            system.debugLog(1, `received in-runtime${data.isSync ? '-sync ' : ''}: ${e.frameRoutingId} [${identity.uuid}]-[${identity.name}] ${JSON.stringify(data)}`);
+            if (data.action === 'publish-message') {
+                //message payload might contain sensitive data, mask it.
+                system.debugLog(1, `received in-runtime${data.isSync ? '-sync ' : ''}: ${e.frameRoutingId} [${identity.uuid}]-[${identity.name}] ${JSON.stringify(data, this.payloadReplacer)}`);
+            } else {
+                system.debugLog(1, `received in-runtime${data.isSync ? '-sync ' : ''}: ${e.frameRoutingId} [${identity.uuid}]-[${identity.name}] ${JSON.stringify(data)}`);
+            }
```
Comment (whyn07m3) [inline src/browser/api_protocol/transport_strategy/elipc_strategy.ts]: How about this:
```js
const replacer = data.action === 'publish-message' ? this.payloadReplacer : null;
system.debugLog(1, `received in-runtime${data.isSync ? '-sync ' : ''}: ${e.frameRoutingId} [${identity.uuid}]-[${identity.name}] ${JSON.stringify(data, replacer)}`);
```

**Comment by weiteho** [comment]
changed the code per suggestion.

File: src/shapes.ts [hunk 4]
```diff
@@ -159,6 +159,7 @@ export interface WindowOptions {
     defaultWidth?: number;
     delay_connection?: boolean;
     description?: string;
+    disableIabSecureLogging ?: boolean;
```
Comment (whyn07m3) [inline src/shapes.ts]: extra space before `?`

**Comment by weiteho** [comment]
weird, visual studio automatically inserted the white space... will check how to disable it...

**Comment by wenjunche** [comment]
@HarsimranSingh this is an app level option

**Comment by HarsimranSingh** [comment]
Sorry I meant window OR app level option. It still needs to be in there.

Also is this option being documented? If so, open a PR for the adapter.

File: src/browser/convert_options.js [hunk 5]
```diff
@@ -62,6 +62,7 @@ function five0BaseOptions() {
         'defaultWidth': 800,
         'delay_connection': false,
         'draggable': false,
+        'disableIabSecureLogging': false,
```
Comment (HarsimranSingh) [inline src/browser/convert_options.js]: :neckbeard: alphabetical order

**Pull request merged**